crates/cli-support/src/js/mod.rs

```diff
@@ -1088,6 +1088,28 @@ impl<'a> Context<'a> {
         Ok(())
     }
 
+    fn expose_get_cached_string_from_wasm(&mut self) -> Result<(), Error> {
+        if !self.should_write_global("get_cached_string_from_wasm") {
+            return Ok(());
+        }
+
+        self.expose_get_object();
+        self.expose_get_string_from_wasm()?;
+
+        self.global("
+            function getCachedStringFromWasm(ptr, len) {
+                if (ptr === 0) {
+                    if (len !== 0) {
```

Comment:
Is this `if` necessary? I don't think ptr/len will ever be both zero, right?

Reply:
It is there to support `Option<String>` and `Option<&str>`.

Using the same function for optional and non-optional basically has no performance cost, but it makes the implementation a *lot* easier, and also reduces the code size.

Reply:
Ah ok I think I missed that, if that's the case I think this could go two ways: it could either still remove this `if` because we can guarantee that `getObject(0)` is always undefined or this I think should have an explicit `return undefined;` to be clear

Reply:
I don't really feel comfortable relying upon `getObject(0) === undefined`, feels like it's leaking internal details.

I guess it's okay if we guarantee that `0` will always be `undefined` (even with `anyref` support).

Reply:
I went ahead and removed the `if`, since I think it's reasonable to guarantee that `getObject(0) === undefined` (both for pragmatic and theoretical reasons, since `undefined` is the closest thing JS has to NUL).